packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java

```diff
@@ -133,9 +134,11 @@ private static String getPathFromRemoteUri(final Context context, final Uri uri)
     InputStream inputStream = null;
     OutputStream outputStream = null;
     boolean success = false;
+    String extension = getImageExtension(context, uri);
+    String extensionWithDot = TextUtils.isEmpty(extension) ? "" : "." + extension;
     try {
       inputStream = context.getContentResolver().openInputStream(uri);
-      file = File.createTempFile("image_picker", "jpg", context.getCacheDir());
+      file = File.createTempFile("image_picker", extensionWithDot, context.getCacheDir());
```

Comment:
what happens to the file path if extensionWithDot is ""

Reply:
nothing, because suffix simply concatenating with prefix. suffix may be null, in which case the suffix .tmp will be used